vcd/resource_vcd_org_user.go

```diff
@@ -0,0 +1,352 @@
+					`Either "password_file" or "password" must be included on creation.`,
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "The Org User's description",
+			},
+			"provider_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  govcd.OrgUserProviderIntegrated,
+				ForceNew: false,
+				Description: "Identity provider type for this this user. One of: 'INTEGRATED', 'SAML', 'OAUTH'. " +
+					"When empty, the default value 'INTEGRATED' is used.",
+			},
+			"full_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "The Org User's full name",
+			},
+			"email_address": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "The Org User's email address",
+			},
+			"telephone": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "The Org User's telephone",
+			},
+			"instant_messaging": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "The Org User's telephone",
+			},
+			"is_enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    false,
+				Description: "True if the user is enabled and can log in.",
+			},
+			"is_group_role": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    false,
+				Description: "True if this user has a group role.",
+			},
+			"is_locked": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: false,
+				Description: "True if the user account has been locked due to too many invalid login attempts. " +
+					"A locked user account can be re-enabled by updating the user with this flag set to false. " +
+					"Only the system can set the value to true.",
+			},
+			"take_ownership": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    false,
+				Description: "Take ownership of user's objects on deletion.",
+			},
+			"deployed_vm_quota": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     10,
+				ForceNew:    false,
+				Description: "Quota of vApps that this user can deploy. A value of 0 specifies an unlimited quota.",
+			},
+			"stored_vm_quota": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     10,
+				ForceNew:    false,
+				Description: "Quota of vApps that this user can store. A value of 0 specifies an unlimited quota.",
+			},
+		},
+	}
+}
+
+func resourceToUserData(d *schema.ResourceData, meta interface{}) (*govcd.OrgUserConfiguration, *govcd.AdminOrg, error) {
```

Comment:
Short comment above the function's purpose would be nice.

Also, why we started to need this kind of function? (I don't believe we have it in other places or it's named differently).

Reply:
We need this function for both Create and Update. Rather than repeating a lot of `Set` functions in two places, this function allows to keep the main functions cleaner.

Reply:
comments added